apps/web/modules/survey/follow-ups/lib/follow-ups.ts

```diff
@@ -0,0 +1,269 @@
+import { getOrganization } from "@/lib/organization/service";
+import { getResponse } from "@/lib/response/service";
+import { getSurvey } from "@/lib/survey/service";
+import { rateLimit } from "@/lib/utils/rate-limit";
+import { validateInputs } from "@/lib/utils/validate";
+import { sendFollowUpEmail } from "@/modules/email";
+import { getSurveyFollowUpsPermission } from "@/modules/survey/follow-ups/lib/utils";
+import { FollowUpResult, FollowUpSendError } from "@/modules/survey/follow-ups/types/follow-up";
+import { z } from "zod";
+import { TSurveyFollowUpAction } from "@formbricks/database/types/survey-follow-up";
+import { logger } from "@formbricks/logger";
+import { ZId } from "@formbricks/types/common";
+import { Result, err } from "@formbricks/types/error-handlers";
+import { TOrganization } from "@formbricks/types/organizations";
+import { TResponse } from "@formbricks/types/responses";
+import { TSurvey } from "@formbricks/types/surveys/types";
+
+const limiter = rateLimit({
+  interval: 60 * 60, // 1 hour
+  allowedPerInterval: 50, // max 50 calls per org per hour
+});
+
+export const evaluateFollowUp = async (
+  followUpId: string,
+  followUpAction: TSurveyFollowUpAction,
+  survey: TSurvey,
+  response: TResponse,
+  organization: TOrganization
+): Promise<FollowUpResult> => {
+  const { properties } = followUpAction;
+  const { to, subject, body, replyTo } = properties;
+  const toValueFromResponse = response.data[to];
+  const logoUrl = organization.whitelabel?.logoUrl || "";
+
+  // Check if 'to' is a direct email address (team member or user email)
+  const parsedEmailTo = z.string().email().safeParse(to);
+  if (parsedEmailTo.success) {
+    try {
+      // 'to' is a valid email address, send email directly
+      await sendFollowUpEmail({
+        html: body,
+        subject,
+        to: parsedEmailTo.data,
+        replyTo,
+        survey,
+        response,
+        attachResponseData: properties.attachResponseData,
+        logoUrl,
+      });
+
+      return {
+        followUpId,
+        status: "success" as const,
+      };
+    } catch (error) {
+      return {
+        followUpId,
+        status: "error",
+        error: error instanceof Error ? error.message : "Something went wrong",
+      };
+    }
+  }
+
+  // If not a direct email, check if it's a question ID or hidden field ID
+  if (!toValueFromResponse) {
+    // throw new Error(`"To" value not found in response data for followup: ${followUpId}`);
+    return {
+      followUpId,
+      status: "error",
+      error: `To value not found in response data for followup: ${followUpId}`,
+    };
+  }
+
+  if (typeof toValueFromResponse === "string") {
+    // parse this string to check for an email:
+    const parsedResult = z.string().email().safeParse(toValueFromResponse);
+    if (parsedResult.data) {
+      // send email to this email address
+      await sendFollowUpEmail({
```

Comment:
I have a bad feeling by having a function exported that we can pass a body, subject,.. to that can likely be misused. We shouldn't have this in the code.
Please come up with a solution that we only have a simple interface exported, e.g. given a responseId, to send a followUp and the function takes care of the rest.
Not sure where this lives. I now see the issue that we have the sendEmail() functionality is in `modules/email`. So maybe the whole followup functionality needs to live there, too?
What do you think?